### PR TITLE
Add linear recall

### PR DESCRIPTION
**Pull request by Sinacam**
## What does this PR do?

Added recall metric.

## Test CLI & API (`bash tests/autotest.sh`)
Test APIs used by main.py.
- [ ] Test Pass
  - (Copy and paste the last outputted line here.)
- [x] Not Applicable (i.e., the PR does not include API changes.)

## Check API Document
If any new APIs are added, please check if the description of the APIs is added to API document. 
- [x] API document is updated ([linear](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/linear.html), [nn](https://www.csie.ntu.edu.tw/~cjlin/libmultilabel/api/nn.html))
- [ ] Not Applicable (i.e., the PR does not include API changes.)

## Test quickstart & API (`bash tests/docs/test_changed_document.sh`)
If any APIs in quickstarts or tutorials are modified, please run this test to check if the current examples can run correctly after the modified APIs are released.

### CODE REVIEW

File: libmultilabel/linear/metrics.py [hunk 1]
```diff
@@ -126,6 +126,40 @@ def reset(self):
...
+            average: Define the reduction that is applied over labels. Currently only "samples" is supported.
+            top_k: Consider only the top k elements for each query.
+        """
+        if average != "samples":
+            raise ValueError("unsupported average")
+
+        _check_top_k(top_k)
+
+        self.top_k = top_k
+        self.score = 0
```
Comment (donglihe-hub) [inline libmultilabel/linear/metrics.py:144]:
```suggestion
        self.add_state(self.score, torch.tensor(0, dtype=torch.float), dist_reduce_fx="sum")
```

File: libmultilabel/linear/metrics.py [hunk 2]
```diff
@@ -126,6 +126,40 @@ def reset(self):
...
+        self.num_sample = 0
```
Comment (donglihe-hub) [inline libmultilabel/linear/metrics.py:145]:
```suggestion
        self.add_state("score", torch.tensor(0, dtype=torch.float), dist_reduce_fx="sum")
        self.add_state("num_sample", torch.tensor(0), dist_reduce_fx="sum")
```

File: libmultilabel/linear/metrics.py [hunk 3]
```diff
@@ -126,6 +126,40 @@ def reset(self):
...
+        top_k_idx = np.argpartition(preds, -self.top_k)[:, -self.top_k :]
+        num_relevant = np.take_along_axis(target, top_k_idx, -1).sum(axis=-1)  # (batch_size, )
+        # zero label instances treated as 0, to be consistent with torchmetrics
+        self.score += np.nan_to_num(num_relevant / target.sum(axis=-1), nan=0.0).sum()
+        self.num_sample += preds.shape[0]
+
+    def compute(self) -> float:
+        return self.score / self.num_sample
+
+    def reset(self):
```
Comment (donglihe-hub) [inline libmultilabel/linear/metrics.py:158]: According to the [documentation](https://torchmetrics.readthedocs.io/en/stable/pages/implement.html), user should not implement reset by theirselves.

File: libmultilabel/linear/metrics.py [hunk 4]
```diff
@@ -126,6 +126,40 @@ def reset(self):
...
+        self.score = 0
+        self.num_sample = 0
```
Comment (donglihe-hub) [inline libmultilabel/linear/metrics.py:160]: According to the [documentation](https://torchmetrics.readthedocs.io/en/stable/pages/implement.html), users should not implement `reset()` by theirselves.

**Review by Eleven1Liu — APPROVED** [review]
